### ISSUE TITLE
[BUG] `FhPlexForecaster.predict` fails if `forecaster` is a grid search model with `refit=False`

### ISSUE THREAD

**Issue by hliebert** [sktime/sktime#4929]
**Describe the bug**
`FhPlexForecaster.predict` fails if `forecaster` is a grid search model with `refit=False`. The exception raised is `NotFittedError`, complaining that the instance of the inner estimator has not been fitted yet.

**To Reproduce**
The code below raises `sktime.exceptions.NotFittedError: This instance of DirectTabularRegressionForecaster has not been fitted yet; please call `fit` first.`

```python
import numpy as np
from sklearn.ensemble import RandomForestRegressor
from sktime.datasets import load_longley
from sktime.forecasting.base import ForecastingHorizon
from sktime.forecasting.compose import make_reduction, FhPlexForecaster
from sktime.forecasting.model_selection import ExpandingWindowSplitter
from sktime.forecasting.model_selection import ForecastingGridSearchCV

y, X = load_longley()
horizon = ForecastingHorizon(np.arange(1, 4), is_relative=True)

random_forest = make_reduction(RandomForestRegressor(), window_length=3, strategy="direct", windows_identical=False)
expanding_window_cv = ExpandingWindowSplitter(fh=horizon, step_length=1)
gscv = ForecastingGridSearchCV(
    forecaster=random_forest,
    cv=expanding_window_cv,
    param_grid={"estimator__max_features": np.linspace(0.1, 0.9, num=2)},
    return_n_best_forecasters=1,
    refit=False,
    verbose=1,
)

forecaster = FhPlexForecaster(forecaster=gscv)

forecaster.fit(y, X, fh=horizon)
forecaster.predict(fh=horizon, X=X)
```

**Expected behavior**
A forecast based on the model fit to the last cross-validation fold.

**Versions**
<details>
System:
    python: 3.8.10 (tags/v3.8.10:3d8993a, May  3 2021, 11:48:03) [MSC v.1928 64 bit (AMD64)]
executable: C:\Program Files\python-3.8.10\python.exe
   machine: Windows-10-10.0.19045-SP0
Python dependencies:
          pip: 21.1.1
       sktime: 0.21.0
      sklearn: 1.2.2
       skbase: 0.4.6
        numpy: 1.23.5
        scipy: 1.10.1
       pandas: 1.5.3
   matplotlib: 3.7.2
       joblib: 1.3.1
  statsmodels: 0.14.0
        numba: 0.57.1
     pmdarima: 2.0.3
      tsfresh: None
   tensorflow: None
tensorflow_probability: None
</details>

**Comment by fkiraly** [comment]
@hliebert, thanks for reporting! This indeed looks like a bug, we'll have a look.

Confirmed, on python 3.11, windows, current `main`.

**Comment by fkiraly** [comment]
simplified example - seems to fail with any forecaster.

```python
import numpy as np
from sktime.datasets import load_longley
from sktime.forecasting.base import ForecastingHorizon
from sktime.forecasting.model_selection import ExpandingWindowSplitter
from sktime.forecasting.model_selection import ForecastingGridSearchCV
from sktime.forecasting.naive import NaiveForecaster

y, X = load_longley()
horizon = ForecastingHorizon(np.arange(1, 4), is_relative=True)

random_forest = make_reduction(RandomForestRegressor(), window_length=3, strategy="direct", windows_identical=False)
expanding_window_cv = ExpandingWindowSplitter(fh=horizon, step_length=1)
gscv = ForecastingGridSearchCV(
    forecaster=NaiveForecaster(),
    cv=expanding_window_cv,
    param_grid={"strategy": ["mean", "last"]},
    return_n_best_forecasters=1,
    refit=False,
    verbose=1,
)

forecaster = FhPlexForecaster(forecaster=gscv)

forecaster.fit(y, X, fh=horizon)
forecaster.predict(fh=horizon, X=X)
```

**Comment by fkiraly** [comment]
I've found the prolem, ut it seems hard to fix.

It looks like the implementer(s) never implemented the logic for `refit=False`, the fitted est forecaster is also not easily obtainable at the place where it should go in this case. The test was also turned off so it was not tested nor detected.

I've made a PR where the gap is highlighted and the test is turned on, for anyone who wants to give this a try:
https://github.com/sktime/sktime/pull/4945

The key difficulty is obtaining the best fitted forecaster, which is not returned from inside the parallelization loop, one would need to write some logic to either pass it through, or store it to an accessible location in `self`.

**Comment by fkiraly** [comment]
@hliebert, see update above.

**Comment by hliebert** [comment]
Thanks for investigating! Good to know this is broken, I was experimenting with sliding windows and refit, but will revert to the default value now.

**Comment by fkiraly** [comment]
@hliebert, I looked into what `sklearn` does, and there's an interesting turn of events now - I should have also remembered a similar discussion with @achieveordie (who I also tag here due to the `refit` issue in classificaiton).

In `sklearn`, if `refit=False`, the grid search will produce the `best_estimator_` attribute, but cannot be used for prediction, i.e., `predict` will fail.

That is, `refit=False` does not use the best estimator in `predict`, it simply doesn't do any fitting of it.

In `sktime` terms, this would be the setting to use the grid search as parameter estimator only. So, failure as above is actually expected.

Now, this turns the action from "fixing" the unexpected behaviour into one of documenting the expected behaviour and raising an error message along the lines, "if you have set `refit=False`, you can't use `predict`".

**Comment by achieveordie** [comment]
In the case of classification, we were delegating certain methods (including predict) to `GridSearchCV` from sklearn (which doesn't support predict with `refit=False`) but that doesn't seem to be the case here.

Quoting from the docstring of `ForecastingGridSearchCV`:
> refit : bool, optional (default=True) 
True = refit the forecaster with the best parameters on the entire data in fit
False = best forecaster remains fitted on the last fold in cv

It seems like the intended behavior was to still support `.predict()` despite `refit=False`?

**Comment by fkiraly** [comment]
yes, but I think the docstring was a "guess", base on a state where there was code and no docstring. It makes sense that the behaviour is actually identical with sklearn.

**Comment by hliebert** [comment]
> @hliebert, I looked into what `sklearn` does, and there's an interesting turn of events now - I should have also remembered a similar discussion with @achieveordie (who I also tag here due to the `refit` issue in classificaiton).
> 
> In `sklearn`, if `refit=False`, the grid search will produce the `best_estimator_` attribute, but cannot be used for prediction, i.e., `predict` will fail.
> 
> That is, `refit=False` does not use the best estimator in `predict`, it simply doesn't do any fitting of it.
> 
> In `sktime` terms, this would be the setting to use the grid search as parameter estimator only. So, failure as above is actually expected.
> 
> Now, this turns the action from "fixing" the unexpected behaviour into one of documenting the expected behaviour and raising an error message along the lines, "if you have set `refit=False`, you can't use `predict`".

Thanks for investigating! I wasn't aware this is behavior is the default in sklearn.

**Comment by fkiraly** [comment]
>I wasn't aware this is the default behavior in sklearn.

Turns out I once understood it but then forgot it again, and got confused about it. Which shows that documentation is key here, as it is not intuitive.
